lms/djangoapps/courseware/models.py

```diff
@@ -287,3 +289,118 @@ def course_allows_nonregistered_access(cls, course_id):
+
+    def __unicode__(self):
+        return "[QueriesSaved] Query {} for {}/{}, {} {}".format(
+            self.id,  # pylint: disable=no-member
+            self.course_id,
+            self.module_state_key,
+            self.get_inclusion_display(),  # pylint: disable=no-member
+            self.filter_on,
+        )
+
+
+class QueriesTemporary(models.Model):
+    """
+    Email Distribution List: Stores individual queries that are in progress. This table is purged periodically
+    """
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    module_state_key = LocationKeyField(max_length=255, db_index=True, db_column='module_id')
+    inclusion = models.CharField(max_length=1, choices=INCLUSIONS)
+    filter_on = models.CharField(max_length=255)
+    created = models.DateTimeField(auto_now_add=True, null=True, db_index=True)
+    entity_name = models.CharField(max_length=255)
+    type = models.CharField(max_length=255)
+    done = models.NullBooleanField()
+
+    def __unicode__(self):
+        return "[QueriesSaved] Query {} for {}/{}, {} {}".format(
+            self.id,  # pylint: disable=no-member
+            self.course_id,
+            self.module_state_key,
+            self.get_inclusion_display(),  # pylint: disable=no-member
+            self.filter_on,
+        )
+
+
+class QueriesStudents(models.Model):
+    """
+    Email Distribution List: Students saved as part of a query. This will get purged periodically so do not query this
+    directly (instead use one of functions that actively makes a query and then returns students). Associate these
+    students with a query in QueriesTemporary
+    """
+    query = models.ForeignKey('QueriesTemporary')
+    student = models.ForeignKey(User, db_index=True)
+    inclusion = models.CharField(max_length=1, choices=INCLUSIONS)
+
+    def __unicode__(self):
+        return "[QueriesStudents] Query {} for {}, {}".format(
+            self.query.id,  # pylint: disable=no-member
+            self.student,
+            self.get_inclusion_display(),  # pylint: disable=no-member
+        )
+
+
+class GroupedTempQueriesSubqueries(models.Model):
+    """
+    Email Distribution List: Saved temporary queries associated with a group per course. Used when loading a
+    saved query and then executing it.
+    """
+    grouped = models.ForeignKey('GroupedQueries')
+    query = models.ForeignKey('QueriesTemporary')
+
+    def __unicode__(self):
+        return "[GroupedQueriesSubqueries] Group {} has Query {}".format(
+            self.grouped.id,  # pylint: disable=no-member
+            self.query.id,  # pylint: disable=no-member
+        )
+
+
+class GroupedQueriesSubqueries(models.Model):
+    """
+    Email Distribution List: Saved queries by instructors per course. Associates permanent queries in QueryiesSaved
+    into GroupedQueries
+    """
+    grouped = models.ForeignKey('GroupedQueries')
+    query = models.ForeignKey('QueriesSaved')
```

Comment:
The GroupedQueriesSubqueries (will rename it SubqueryForGroupedQuery) are all associated with a GroupedQuery, which has a title and time of creation. I didn't think it was necessary to add storing the user who created it since this tool is limited to instructors and possibly course staff. what do you think?